send-transaction-service/src/transaction_client.rs

```diff
@@ -243,84 +231,67 @@ impl TpuClientNextClient {
         my_tpu_address: SocketAddr,
         tpu_peers: Option<Vec<SocketAddr>>,
         leader_info: Option<T>,
-        leader_forward_count: u64,
+        leader_forward_count: usize,
         identity: Option<&Keypair>,
         bind_socket: UdpSocket,
         cancel: CancellationToken,
     ) -> Self
     where
         T: TpuInfoWithSendStatic + Clone,
     {
-        // The channel size represents 8s worth of transactions at a rate of
-        // 1000 tps, assuming batch size is 64.
-        let (sender, receiver) = mpsc::channel(128);
-
-        let (update_certificate_sender, update_certificate_receiver) = watch::channel(None);
-
         let leader_info_provider = CurrentLeaderInfo::new(leader_info);
         let leader_updater: SendTransactionServiceLeaderUpdater<T> =
             SendTransactionServiceLeaderUpdater {
                 leader_info_provider,
                 my_tpu_address,
                 tpu_peers,
             };
-        let config = Self::create_config(bind_socket, identity, leader_forward_count as usize);
-
-        let scheduler = ConnectionWorkersScheduler::new(
-            Box::new(leader_updater),
-            receiver,
-            update_certificate_receiver,
-            cancel.clone(),
-        );
-        // leaking handle to this task, as it will run until the cancel signal is received
-        runtime_handle.spawn(scheduler.get_stats().report_to_influxdb(
-            "send-transaction-service-TPU-client",
-            METRICS_REPORTING_INTERVAL,
-            cancel.clone(),
-        ));
-        let _handle = runtime_handle.spawn(scheduler.run(config));
+
+        let (sender, client) = ClientBuilder::with_leader_updater(Box::new(leader_updater))
+            .runtime_handle(runtime_handle.clone())
+            .bind_socket(bind_socket)
+            .identity(identity)
+            .leader_send_fanout(leader_forward_count)
+            .max_cache_size(MAX_CONNECTIONS)
+            // The channel size represents 8s worth of transactions at a rate of
+            // 1000 tps, assuming batch size is 64.
+            .sender_capacity(128)
+            .worker_channel_size(64)
+            .max_reconnect_attempts(4)
+            .report({
+                |stats: Arc<SendTransactionStats>, cancel: CancellationToken| async move {
+                    stats
+                        .report_to_influxdb(
+                            "send-transaction-service-TPU-client",
+                            METRICS_REPORTING_INTERVAL,
+                            cancel.clone(),
+                        )
+                        .await
+                }
+            })
```

Comment:
*minor*

Might be an overkill, but I wonder if `SendTransactionStats` could provide a method that constructs with kind of function, given the DB key.